In [2]:
import pandas as pd

data = pd.read_csv('insurance_with_bmi.csv')
# Select only the numeric columns for correlation calculation
numeric_columns = data[['age', 'bmi', 'children', 'charges']]

# Compute the correlation matrix for the numeric columns in the dataset
correlation_matrix = numeric_columns.corr()

# Display the correlation matrix
correlation_matrix

age       bmi  children   charges
age       1.000000  0.109272  0.042469  0.299008
bmi       0.109272  1.000000  0.012759  0.198341
children  0.042469  0.012759  1.000000  0.067998
charges   0.299008  0.198341  0.067998  1.000000

Penjelasan Hasil:
Setelah menjalankan script di atas, berikut adalah penjelasan dari correlation matrix yang dihitung:

Umur (age) memiliki korelasi positif dengan charges (0.299), menunjukkan bahwa semakin tua seseorang, biaya asuransi mereka cenderung meningkat secara moderat.
BMI juga berkorelasi positif dengan charges (0.198), menunjukkan bahwa orang dengan BMI lebih tinggi cenderung memiliki biaya asuransi yang lebih besar, meskipun korelasinya lebih lemah dibandingkan dengan umur.
Jumlah anak (children) memiliki korelasi sangat lemah dengan charges (0.068), menunjukkan bahwa jumlah anak hampir tidak mempengaruhi biaya asuransi.
Kesimpulannya, umur dan BMI adalah faktor-faktor yang paling berpengaruh terhadap biaya asuransi, dengan umur memiliki pengaruh yang lebih kuat.